internal/pkg/skuba/cni/cilium_test.go

```diff
@@ -0,0 +1,345 @@
+		})
+	}
+}
+
+func Test_CreateOrUpdateCiliumConfigMap(t *testing.T) {
+	cmBase := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kubeadm-config",
+			Namespace: metav1.NamespaceSystem,
+		},
+		Data: map[string]string{"ClusterStatus": "apiEndpoints:\n  master:\n    advertiseAddress: 1.2.3.4\n    bindPort: 80\napiVersion: kubeadm.k8s.io/v1beta1\nkind: ClusterStatus\n"},
+	}
+
+	cmDefaultNamespace := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake",
+			Namespace: metav1.NamespaceDefault,
+		},
+		Data: map[string]string{"hello": "world"},
+	}
+
+	tests := []struct {
+		clientset   *fake.Clientset
+		configmap   *corev1.ConfigMap
+		errExpected bool
+		errMessage  string
+		name        string
+		namespace   string
+	}{
+		{
+			name:        "should create or update cilium configmap",
+			clientset:   fake.NewSimpleClientset(),
+			configmap:   cmBase,
+			errExpected: false,
+		},
+		{
+			name:        "should fail when kubeadm-config configmap not exist",
+			clientset:   fake.NewSimpleClientset(),
+			configmap:   cmDefaultNamespace,
+			errExpected: true,
+			errMessage:  "unable to get api endpoints: could not retrieve the kubeadm-config configmap to get apiEndpoints: configmaps \"kubeadm-config\" not found",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			tt.clientset.CoreV1().ConfigMaps("kube-system").Create(tt.configmap)
+
+			err := CreateOrUpdateCiliumConfigMap(tt.clientset)
+
+			if tt.errExpected {
+				if err == nil {
+					t.Errorf("error expected on %s, but no error reported", tt.name)
+					return
+				}
+				if err.Error() != tt.errMessage {
+					t.Errorf("(%v) expected on %s, but different error message reported (%v)", tt.errMessage, tt.name, err.Error())
+					return
+				}
+			} else if !tt.errExpected {
+				if err != nil {
+					t.Errorf("error not expected on %s, but an error was reported (%s)", tt.name, err.Error())
+					return
+				}
+
+				dataGet, err := tt.clientset.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get("cilium-config", metav1.GetOptions{})
+				if err != nil {
+					t.Errorf("error not expected on %s, but an error was reported (%s)", tt.name, err.Error())
+					return
+				}
+
+				dataExpected := map[string]string{"debug": "false", "disable-ipv4": "false", "etcd-config": "endpoints:\n- https://1.2.3.4:2379\nca-file: /tmp/cilium-etcd/ca.crt\ncert-file: /tmp/cilium-etcd/tls.crt\nkey-file: /tmp/cilium-etcd/tls.key\n"}
+				if !reflect.DeepEqual(dataGet.Data, dataExpected) {
+					t.Errorf("(%v) expected on %s, but different configmap data reported (%v)", dataExpected, tt.name, dataGet.Data)
+					return
+				}
+			}
+		})
+	}
+}
+
+func Test_GetCiliumImage(t *testing.T) {
+	tests := []struct {
+		name         string
+		repoExpected string
+	}{
+		{
+			name:         "should get cilium image repository",
+			repoExpected: "registry.suse.de/devel/caasp/4.0/containers/containers/caasp/v4/cilium",
```

Comment:
Not sure testing the image repository is needed or not.

And we have different image repo URL by build flag:
https://github.com/SUSE/skuba/blob/master/pkg/skuba/development_constants.go#L1
https://github.com/SUSE/skuba/blob/master/pkg/skuba/staging_constants.go#L1
https://github.com/SUSE/skuba/blob/master/pkg/skuba/release_constants.go#L1

So this test would pass only when build flag is not staging or not release.
I am not sure when we release the package to staging or release, it will run the unit test or not.

Reply:
Removed.